Signal/src/ViewControllers/DebugUI/DebugUIDiskUsage.m

```diff
@@ -0,0 +1,299 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "DebugUIDiskUsage.h"
+#import "OWSTableViewController.h"
+#import "Signal-Swift.h"
+#import <SignalServiceKit/TSDatabaseView.h>
+#import <SignalServiceKit/TSInteraction.h>
+#import <SignalServiceKit/TSStorageManager.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation DebugUIDiskUsage
+
+#pragma mark - Logging
+
++ (NSString *)tag
+{
+    return [NSString stringWithFormat:@"[%@]", self.class];
+}
+
+- (NSString *)tag
+{
+    return self.class.tag;
+}
+
+#pragma mark - Factory Methods
+
+- (NSString *)name
+{
+    return @"Orphans & Disk Usage";
+}
+
+- (nullable OWSTableSection *)sectionForThread:(nullable TSThread *)thread
+{
+    return [OWSTableSection sectionWithTitle:self.name
+                                       items:@[
+                                           [OWSTableItem itemWithTitle:@"Audit & Log"
+                                                           actionBlock:^{
+                                                               [DebugUIDiskUsage auditWithoutCleanup];
+                                                           }],
+                                           [OWSTableItem itemWithTitle:@"Audit & Clean Up"
+                                                           actionBlock:^{
+                                                               [DebugUIDiskUsage auditWithCleanup];
+                                                           }],
+                                           [OWSTableItem itemWithTitle:@"Save All Attachments"
+                                                           actionBlock:^{
+                                                               [DebugUIDiskUsage saveAllAttachments];
+                                                           }],
+                                           [OWSTableItem itemWithTitle:@"Delete Messages older than 3 Months"
+                                                           actionBlock:^{
+                                                               [DebugUIDiskUsage deleteOldMessages_3Months];
+                                                           }],
+                                       ]];
+}
+
++ (void)auditWithoutCleanup
+{
+    [self auditAndCleanup:NO];
+}
+
++ (void)auditWithCleanup
+{
+    [self auditAndCleanup:YES];
+}
+
++ (void)auditAndCleanup:(BOOL)shouldCleanup
+{
+    NSString *attachmentsFolder = [TSAttachmentStream attachmentsFolder];
+    DDLogError(@"attachmentsFolder: %@", attachmentsFolder);
+
+    __block int fileCount = 0;
+    __block long long totalFileSize = 0;
+    NSMutableSet *diskFilePaths = [NSMutableSet new];
```

Comment:
Could I convince you to move this implementation to the orphan file cleaner?

motivation:

1. That class is broken now that we have  subdirectories in our attachment path.
2. It seems like they'll be a day when we use this outside of debugUI.